code/modules/vending/_vending.dm

```diff
@@ -387,42 +392,23 @@ GLOBAL_LIST_EMPTY(vending_products)
 				else
 					to_chat(user, "<span class='warning'>There's nothing to restock!</span>")
 			return
-	if(compartmentLoadAccessCheck(user))
-		if(canLoadItem(I))
-			loadingAttempt(I,user)
-			updateUsrDialog() //can't put this on the proc above because we spam it below
-
-		if(istype(I, /obj/item/storage/bag)) //trays USUALLY
-			var/obj/item/storage/T = I
-			var/loaded = 0
-			var/denied_items = 0
-			for(var/obj/item/the_item in T.contents)
-				if(contents.len >= MAX_VENDING_INPUT_AMOUNT) // no more than 30 item can fit inside, legacy from snack vending although not sure why it exists
-					to_chat(user, "<span class='warning'>[src]'s chef compartment is full.</span>")
-					break
-				if(canLoadItem(the_item) && loadingAttempt(the_item,user))
-					SEND_SIGNAL(T, COMSIG_TRY_STORAGE_TAKE, the_item, src, TRUE)
-					loaded++
-				else
-					denied_items++
-			if(denied_items)
-				to_chat(user, "<span class='warning'>[src] refuses some items!</span>")
-			if(loaded)
-				to_chat(user, "<span class='notice'>You insert [loaded] dishes into [src]'s chef compartment.</span>")
-				updateUsrDialog()
+	if(canLoadItem(I) && !custom)
+		loadingAttempt(I,user)
+		updateUsrDialog() //can't put this on the proc above because we spam it below
```

Comment:
where did this code go

Reply:
chef thing has been replaced by custom vendor

Reply:
this is not exclusive to chef items.

Reply:
actually they are in practice but the ability to bar certain people to loading from a vendor that isn't a personal one seems like a valid usecase.